nautobot/docs/development/best-practices.md

```diff
@@ -92,7 +92,37 @@ class UserFilter(NautobotFilterSet):
 - It is acceptable that default filter mappings **may** need to be overridden with custom filter declarations, but [`filter_overrides`](https://django-filter.readthedocs.io/en/stable/ref/filterset.html#customise-filter-generation-with-filter-overrides) (see below) should be used as a first resort.
 - Custom filter definitions **must not** shadow the name of an existing model field if it is also changing the type.
     - For example `DeviceFilterSet.interfaces` is a `BooleanFilter` that is shadowing the `Device.interfaces` related manager. This introduces problems with automatic introspection of the filterset and this pattern **must** be avoided.
-- Custom filter definitions **may** shadow the name of an existing field IF this is to adapt a filter for a human-readable field value (such as `slug`) on a related field vs. the `pk` (the default for most related fields). This pattern still relies on accessing the related field by name and then traversing into the relationship using a nested lookup.
+- For foreign-key related fields, **on existing core models in the v1.3 release train**: 
+    - The field **should** be shadowed, replacing the PK filter with a lookup-based on a more human-readable value (typically `slug`, if available). 
+    - A PK-based filter **should** be made available as well, generally with a name suffixed by `_id`. For example:
+
+
+```python
+    provider = django_filters.ModelMultipleChoiceFilter(
+        field_name="provider__slug",
+        queryset=Provider.objects.all(),
+        to_field_name="slug",
+        label="Provider (slug)",
+    )
+    provider_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=Provider.objects.all(),
+        label="Provider (ID)",
+    )
+```
+
+- For foreign-key related fields on **new core models for v1.4 or later:**
+    - The field **must** be shadowed utilizing a hybrid `NaturalKeyMultipleChoiceFilter` which will automatically try to lookup by UUID or `slug` depending on the value of the incoming argument (e.g. UUID vs. slug). 
+    - Using this filter by default, `provider` would be a `pk` (UUID) field, whereas ithis field will automatically support both input values for `slug` or `pk`.
```

Comment:
Not sure I parse this sentence. Is "this filter" referring to `NaturalKeyMultipleChoiceFilter` or the default (pk-only) filtering?